### PR TITLE
[O2B-1344] Add data pass status

### PR DESCRIPTION
**Pull request by xsalonx**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [x] issue has "Fix version" assigned
- [x] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for users:
- NA

Notable changes for developers:
- NA

Changes made to the database:
- Removed data_pass_versions column 'deleted_from_monalisa' and added data_pass_version_status_history table

### CODE REVIEW

File: lib/server/monalisa-synchronization/MonAlisaSynchronizer.js [hunk 1]
```diff
@@ -130,41 +142,75 @@ class MonAlisaSynchronizer {
...
             attributes: ['id', 'description', 'lastSeen'],
-            where: { deletedFromMonAlisa: false },
+            include: 'statusHistory',
+            order: [['statusHistory', 'createdAt', 'ASC']],
         });
 
-        return Object.fromEntries(dataPassesLastSeen.map(({ description, lastSeen, id }) => [description, { lastSeen, id }]));
+        return Object.fromEntries(dataPassesLastSeen.map(({ description, lastSeen, id, statusHistory }) => [
+            description,
+            { lastSeen, id, status: statusHistory.slice(-1)[0].status },
```
Comment (martinboulais) [inline lib/server/monalisa-synchronization/MonAlisaSynchronizer.js:158]: I think `statusHistory[statusHistory.length - 1]` is the conventional way to go

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: lib/database/adapters/DataPassVersionAdapter.js [hunk 2]
```diff
@@ -49,7 +53,9 @@ class DataPassVersionAdapter {
             reconstructedEventsCount,
             outputSize,
             lastSeen,
-            deletedFromMonAlisa,
+
+            deletedFromMonAlisa: (statusHistory ?? [])
+                .map(this.dataPassVersionStatusAdapter.toEntity).slice(-1)[0]?.status === DataPassVersionStatus.DELETED,
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/database/adapters/DataPassVersionAdapter.js]: refactored

File: lib/database/adapters/DataPassVersionStatusAdapter.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+/**
+ * DataPassVersionStatusAdapter
+ */
+class DataPassVersionStatusAdapter {
+    /**
+     * Constructor
+     */
+    constructor() {
+        this.toEntity = this.toEntity.bind(this);
+        this.dataPassVersionStatusAdapter = null;
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/database/adapters/DataPassVersionStatusAdapter.js:23]: applied

File: lib/server/monalisa-synchronization/MonAlisaSynchronizer.js (hunk 1 above)
Comment (xsalonx) [reply lib/server/monalisa-synchronization/MonAlisaSynchronizer.js]: applied

**Review by martinboulais — APPROVED** [review]

**Pull request merged**